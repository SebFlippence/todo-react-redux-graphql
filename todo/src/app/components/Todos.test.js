import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { stub, sandbox } from 'sinon';
import * as actionQuery from '../actions/query';

import Todos from './Todos';

let boxed;
let getTodosStub;

beforeEach(() => {
  boxed = sandbox.create();
  getTodosStub = boxed.stub(actionQuery, 'getTodos');
});

afterEach(() => {
  boxed.restore();
});

it('renders an empty todos list', () => {
  let dispatchStub = stub();
  const component = renderer.create(
    <Todos
      dispatch={dispatchStub}
      store={{
        fetchInProgress: false,
        allTodos: [],
      }}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(dispatchStub.calledOnce).toBeTruthy();
  expect(getTodosStub.calledOnce).toBeTruthy();
});

it('renders an existing todos list', () => {
  let dispatchStub = stub();
  const component = renderer.create(
    <Todos
      dispatch={dispatchStub}
      store={{
        fetchInProgress: false,
        allTodos: [{ action: 'foo', completed: false, id: 123 }],
      }}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(dispatchStub.calledOnce).toBeTruthy();
  expect(getTodosStub.calledOnce).toBeTruthy();
});
