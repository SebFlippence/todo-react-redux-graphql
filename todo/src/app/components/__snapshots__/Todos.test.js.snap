// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an empty todos list 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="Add new todo..."
      type="text"
    />
     
    <input
      type="submit"
      value="Add"
    />
  </form>
</div>
`;

exports[`renders an existing todos list 1`] = `
<div>
  <p>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
     
    <span
      onDoubleClick={[Function]}
    >
      foo
       
    </span>
    <button
      onClick={[Function]}
    >
      <span
        aria-label="Delete"
        role="img"
      >
        ‚ùå
      </span>
    </button>
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="Add new todo..."
      type="text"
    />
     
    <input
      type="submit"
      value="Add"
    />
  </form>
</div>
`;
